src/main/java/seedu/typed/model/ModelManager.java

```diff
@@ -149,8 +149,9 @@ public String toString() {
         @Override
         public boolean run(ReadOnlyTask task) {
             return nameKeyWords.stream()
-                    .filter(keyword -> StringUtil.containsWordIgnoreCase(task.getName().getValue(), keyword)).findAny()
-                    .isPresent();
+                    .filter(keyword -> StringUtil
+                    .containsFuzzyWordIgnoreCase(task.getName().getValue(), keyword))
```

Comment:
I think there's some issues with regards to dependencies (but I've never understood them). In ModelManager you need a function from StringUtil which leads to FindCommandParser's function. Then again, it was FindCommandParser which got you into ModelManager in the first place.
I suggest that Parsers in general do their job as parsers: to parse a string and break it into tokens. Extract out the fuzzy find method into another class which does the job of finding and sorting.